modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/common/util/WindowsAPIUtils.java

```diff
@@ -207,4 +210,34 @@ public static void startTenantFlow(String userName) {
         privilegedCarbonContext.setTenantDomain(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
         privilegedCarbonContext.setUsername(userName);
     }
+
+    /**
+     * Get the host name from the carbon.xml file.
+     *
+     * @return
+     */
+    public static String getHostName() {
+        ServerConfiguration serverConfiguration = CarbonUtils.getServerConfiguration();
+        String[] hostNameCharacters = serverConfiguration.getProperties(PluginConstants.Discovery.HOST_NAME);
+        StringBuilder builder = new StringBuilder();
+        for (String s : hostNameCharacters) {
+            builder.append(s);
+        }
+        return builder.toString();
+    }
+
+    /**
+     * This method is used to get tenant configurations.
+     *
+     * @return List of Configurations entries.
+     * @throws DeviceManagementException
+     */
+    public static List<ConfigurationEntry> getTenantConfigurationData() throws DeviceManagementException {
+        if (WindowsAPIUtils.getTenantConfiguration() != null) {
```

Comment:
In the meantime, why are we exposing two public static methods to get tenant configuration data (i.e. WindowsAPIUtils.getTenantConfiguration(), WindowsAPIUtils.getTenantConfigurationData()) when it can be achieved just by using one method?